frontera/worker/db.py

```diff
@@ -137,51 +137,57 @@ def consume_incoming(self, *args, **kwargs):
-                    _, request, error = msg
-                    logger.debug("Request error %s", request.url)
-                    if b'jid' not in request.meta or request.meta[b'jid'] != self.job_id:
+                    if type == 'links_extracted':
+                        _, request, links = msg
+                        logger.debug("Links extracted %s (%d)", request.url, len(links))
+                        if b'jid' not in request.meta or request.meta[b'jid'] != self.job_id:
+                            continue
+                        self._backend.links_extracted(request, links)
                         continue
-                    self._backend.request_error(request, error)
-                    continue
-                if type == 'offset':
-                    _, partition_id, offset = msg
-                    producer_offset = self.spider_feed_producer.get_offset(partition_id)
-                    if producer_offset is None:
+                    if type == 'request_error':
+                        _, request, error = msg
+                        logger.debug("Request error %s", request.url)
+                        if b'jid' not in request.meta or request.meta[b'jid'] != self.job_id:
+                            continue
+                        self._backend.request_error(request, error)
                         continue
-                    else:
-                        lag = producer_offset - offset
-                        if lag < 0:
-                            # non-sense in general, happens when SW is restarted and not synced yet with Spiders.
+                    if type == 'offset':
+                        _, partition_id, offset = msg
+                        producer_offset = self.spider_feed_producer.get_offset(partition_id)
+                        if producer_offset is None:
                             continue
-                        if lag < self.max_next_requests or offset == 0:
-                            self.spider_feed.mark_ready(partition_id)
                         else:
-                            self.spider_feed.mark_busy(partition_id)
+                            lag = producer_offset - offset
+                            if lag < 0:
+                                # non-sense in general, happens when SW is restarted and not synced yet with Spiders.
+                                continue
+                            if lag < self.max_next_requests or offset == 0:
+                                self.spider_feed.mark_ready(partition_id)
+                            else:
+                                self.spider_feed.mark_busy(partition_id)
+                        continue
+                    logger.debug('Unknown message type %s', type)
+                except Exception as exc:
+                    logger.exception(exc)
+                    if logger.isEnabledFor(logging.DEBUG):
+                        logger.debug("Message caused the error %s", str(msg))
```

Comment:
Can we rephrase this as `An error occurred in the following message: %s`?

`Message caused the error` is a little confusing to me